src/object/cli_obj.c

```diff
@@ -2221,10 +2235,8 @@ shard_io(tse_task_t *task, struct shard_auxi_args *shard_auxi)
 	rc = obj_shard_open(obj, shard_auxi->shard, shard_auxi->map_ver,
 			    &obj_shard);
 	if (rc != 0) {
-		/* skip a failed target */
-		if (rc == -DER_NONEXIST)
-			rc = 0;
```

Comment:
During rebuild/migration, it will first try to fetch on the leader, if it fails, it will retry on non-leader, until all replicas fails. 

So if it ignore the failure here, it will be hard for migrate module to drive the above process.

Reply:
I assume it is just want to ignore the failure for the outside user, but it seems unnecessary and also unnecessary.